asserts/device_asserts.go

```diff
@@ -276,12 +276,13 @@ func (ser *Serial) Timestamp() time.Time {
 // TODO: implement further consistency checks for Serial but first review approach
 
 func assembleSerial(assert assertionBase) (Assertion, error) {
-	err := checkAuthorityMatchesBrand(&assert)
-	if err != nil {
-		return nil, err
+	authorityID := assert.AuthorityID()
+	brand := assert.HeaderString("brand-id")
+	if brand != authorityID && authorityID != "generic" {
+		return nil, fmt.Errorf(`authority-id and brand-id must match, serial assertions are expected to be signed by the brand or the "generic" authority: %q != %q`, authorityID, brand)
```

Comment:
The second part of the sentence contradicts the first one. Perhaps something like:

`serial assertions must be generic or signed by the brand (%q) instead of %q`